src/player.cpp

```diff
@@ -12251,26 +12251,36 @@ int player::encumb( body_part bp ) const
 int player::encumb(body_part bp, double &layers, int &armorenc) const
 {
     int ret = 0;
-    int layer[MAX_CLOTHING_LAYER] = { };
+    // First is the total of encumbrance on the given layer,
+    // and second is the highest encumbrance of any one item on the layer.
+    std::array<std::pair<int, int>, MAX_CLOTHING_LAYER> layer = {{{0,0},{0,0},{0,0},{0,0},{0,0}}};
```

Comment:
It's certainly not the first time I've seen this cause this exact same problem, and I'm not convinced it added value in any of the cases. I'd be okay with turning off the stupid warning.